### DISCUSSION TITLE
coexist with basic exim4 server mail

### DISCUSSION THREAD

**Discussion by Waldorf3** [docker-mailserver/docker-mailserver#3588] [Q&A] [answered]
The Debian Bookworm host running this container needs to be able to send standard system related notifications, so I have installed mailutils and exim4.  Using "dpkg-reconfigure exim4-config" I set it up with first option, standard internet server, but although I don't expect any incoming mail it insist on at least listening on 127.0.0.1:25.  It keeps complaining about this being busy because docker-mailserver is using it.  Not a big problem, except it keeps trying, and if it succeeds docker-mailserver will not be able to use it, and that will be a problem.

For now I change the bind line in docker-compose to bind explicitly to the server IP address instead of the usual 25:25, but I am not sure if that will be a problem for docker-mailserver. Maybe some of the subprocesses try to connect to 127.0.0.1:25 - and will then get the host's exim4 server instead.

How to best deal with this?

**Comment by polarathene** [comment, marked as answer]
> Maybe some of the subprocesses try to connect to `127.0.0.1:25` - and will then get the host's exim4 server instead.

- Each container has an IP assigned to it from private address (_unless you use host mode networking_). This should be unique to the container, no overlap to worry about and you can set an explicit IP within that subnet for reliability if you need to.
- Likewise each container has its own loopback `127.0.0.1`, unrelated to the host (_unless you use host mode networking_).

You should be fine by choosing to restrict the interfaces your container binds ports to. The default is `0.0.0.0` (all) for convenience. If you do not need to reach the container via `localhost` / `127.0.0.1` on your Docker host, there is no need to bind to it 👍
